CT2/MD5/Common.cs

```diff
@@ -0,0 +1,53 @@
+namespace MyMD5;
+
+public class DirectoryHashCommon
+{
+    public enum ThreadMode { SingleThread, MultiThread }
+    public class DirectoryHash 
+    {
+        public long Time { get; }
+        public byte[] Hash { get; }
+        public string Path { get; }
+        public ThreadMode Mode { get; }
+        public DirectoryHash(string path, long time, byte[] hash, ThreadMode mode)
+        {
+            Path = path;
+            Time = time;
+            Hash = hash;
+            Mode = mode;
+        }
+    }
```

Comment:
Это выглядит прямо как классический record:
```suggestion
    public record DirectoryHash(string Path, long Time, byte[] Hash, ThreadMode Mode);
```
VS Code с C# DevKit даже само предлагает заменить.